src/api/java/com/minecolonies/api/compatibility/CompatibilityManager.java

```diff
@@ -732,4 +735,34 @@ private void discoverModCompat()
             Compatibility.dynamicTreesCompat = new DynamicTreeCompat();
         }
     }
+
+    /**
+     * Gets a function returning all the possible foods for a cook building view.
+     * 
+     * @return the function.
+     */
+    public static Function<IBuildingView, Set<ItemStorage>> getAllFoodsFunction()
+    {
+        return (buildingView) ->
+        {
+            Set<ItemStorage> edibles = IColonyManager.getInstance().getCompatibilityManager().getEdibles();
+            edibles.removeIf(item -> ISCOOKABLE.test(item.getItemStack()));
+            return edibles;
+        };
+    }
+
+    /**
+     * Gets a function returning all the possible flowers for a beekeeper building view.
+     * 
+     * @return the function.
+     */
+    public static Function<IBuildingView, Set<ItemStorage>> getAllFlowersFunction()
+    {
+        return (buildingView) ->
```

Comment:
Ohh, I thought you'd make it (buildingView) -> CompatibilityManager.getAllFlowers()

Reply:
I can change it if you prefer it that way, but I thought if i make a new method for this anyways I might as well make the initializer line as short as possible.

Reply:
Yeah, I want it clear that is expects a function lambda, but make the lambda short

Reply:
Ok, i'll Change that later.

Reply:
I haven't done much testing after this change, but I don't see any way for me to mess this up in a way that makes basic functionality work, but kills something else.
So this should be done :)